contents/revisionControl/bestPracticesGit.md

```diff
@@ -0,0 +1,311 @@
+# Best Practices with `git`
+
+Author(s): [Darren Wee](https://github.com/darrenwee)
+
+- [Introduction](#introduction)
+- [Best Practices](#best-practices)
+    - [Write Good Commit Messages](#write-good-commit-messages)
+        - [What Constitutes a Good Commit Message](#what-constitutes-a-good-commit-message)
+        - [Set Up Your Editor for Commit Messages](#set-up-your-editor-for-commit-messages)
+    - [Always Commit Functional Code](#always-commit-functional-code)
+        - [Stashing](#stashing)
+    - [One Logical Change to One Commit](#one-logical-change-to-one-commit)
+    - [Hide the Sausage Making](#hide-the-sausage-making)
+        - [How to Hide the Sausage](#how-to-hide-the-sausage)
+    - [Respect Published History](#respect-published-history)
+    - [Keep Up To Date](#keep-up-to-date)
+        - [Working with Remotes](#working-with-remotes)
+        - [Rebase versus Merging](#rebase-versus-merging)
+- [Resources](#resources)
+    - [Further Reading](#further-reading)
+
+
+# Introduction
+[`git`](https://git-scm.com/) is a popular source code management tool and commonly used in many open-source projects, especially those on [GitHub](https://github.com).
+
+# Best Practices
+`git` can be an incredibly useful tool for collaboration or it can be a terrible headache. Best practices exist in order to create a common understanding between users so that the latter does not happen.
+
+Best practices are guidelines that are mostly sensible, but are still guidelines. You can always choose to ignore them although best if you have a compelling reason to do so.
+
+---
+
+## Write Good Commit Messages
+Good commit messages can help:
+- reviewers to understand the high-level changes in your pull request/patch
+- others to understand your reasoning behind the changes made by that commit,
+    - while reviewing your code,
+    - while figuring out why a piece of code that is five years old is that way,
```

Comment:
the comma here is kinda weird.

Reply:
I will revise the structure to clean this part up and remove the redundant `while` and commas.